Duplicati/Server/WebServer/RESTMethods/Backup.cs

```diff
@@ -65,7 +65,9 @@ private void SearchFiles(IBackup backup, string filterstring, RequestInfo info)
                 result[n.Name] = n.Value;
 
             result["Filesets"] = r.Filesets;
-            result["Files"] = r.Files;
+            result["Files"] = r.Files
+                .OrderByDescending(f => f.Path.EndsWith(Path.DirectorySeparatorChar.ToString(), StringComparison.Ordinal))

```

Comment:
How does this affect cross-system restores, e.g., a Windows backup being restored to a Linux machine?

Also, there's a [cached version](https://github.com/duplicati/duplicati/blob/v2.0.5.104-2.0.5.104_canary_2020-03-25/Duplicati/Library/Common/IO/Util.cs#L27) of the directory separator to avoid repeated conversions from `char` to `string`.  I'm not sure how much it saves in terms of performance, but it is being used elsewhere.

Reply:
I tested restoring Linux data on a Windows machine, and basically the first sort doesn't work because of the wrong slash.  The end result was the files and folders were sorted alphabetically but mixed.

One possible solution is to check if the string EndsWith either forward slash OR backslash, regardless of platform.  What do you think?

Reply:
Went ahead and pushed a commit with that change.  Tested and it works even when restoring from an alternate platform.

Reply:
There is the case where several Linux filesystems (e.g., ext4) allow `\` in filenames, so technically there could be a file whose name ends with `\`.  For example, using revision 7a4d90f4af329409e47718bdb25d21a24853693f, I have a file called `weird\` that is sorted as if it was a folder:

![image](https://user-images.githubusercontent.com/17345532/77869981-131fd180-71f5-11ea-880b-7d014771a079.png)

Somehow, the restore tree knows that this is a file.  Can you test backing up a file that ends with `\` on Linux, and see how the restore dialog appears in Windows?  Is it rendered as a file or as a folder?

Reply:
Very interesting. I'll do some testing tomorrow.

Reply:
Ok, backed up some test data on Linux that includes a filename with a trailing backslash.  Doesn't surprise me that on Linux this has the regular file icon, since a backslash is not the platform's path separator.

But I was surprised to see that on Windows this Linux backup data is displayed the same way.  The file is shown with a file icon.  It's also NOT grouped with the folders:
![Capture](https://user-images.githubusercontent.com/39164691/77942340-53706580-7270-11ea-9988-fece943bff5a.PNG)

(Out of curiosity I tried restoring it on Windows. The restore was successful and the file was named afile.txt_, which I think is a good compromise.)

This file shows up with two backslashes when debugging, which makes sense to me.
![Capture2](https://user-images.githubusercontent.com/39164691/77943169-a8f94200-7271-11ea-9e09-ed978c45efcb.PNG)

I would have expected this to cause the file to be grouped with folders though.

How about this idea - instead of going by the trailing slash/backslash, we go by the Size property. Looks like Duplicati sets this to `-1` for folders.  Do you think that'd be a more reliable method?

Reply:
Sorry, I made a mistake in my first screen shot. That was on Windows but Duplicati didn't have this change applied.  When I have this pull request applied it definitely sorts `afile.txt\` up with the folder group.

I'd like to test going by size = -1 unless you think that's a bad idea.

Reply:
The use of `-1` as the size is littered throughout the code, and its indication of a path being a folder seems more like a side effect rather than an enforced standard.  I would prefer not to code against this.

The UI code attempts to [guess](https://github.com/duplicati/duplicati/blob/v2.0.5.104-2.0.5.104_canary_2020-03-25/Duplicati/Server/webroot/ngax/scripts/controllers/RestoreController.js#L175) the path separator by inspecting the first character.  If it's `/`, then the files came from a non-Windows backup, and hence a path ending with `/` should indicate a folder.  If the first character is not `/`, then the files came from a Windows backup  and hence a path ending with `\` should indicate a folder.

Do you think we can apply the same logic here?

Reply:
Perhaps... give me some time to mess around with this.  Thanks!